### DISCUSSION TITLE
CameraView and device orientation

### DISCUSSION THREAD

**Discussion by RcBrgr** [CommunityToolkit/Maui#1967] [Q&A]
I have played around a bit with CameraView and also captured images in landscape orientation but the images are rotated. Do I need to make some configuration like changing the resolution or other configuration?

**Comment by bijington** [comment]
What platform are you seeing this on?

**Reply by RcBrgr** [reply]
Android - Samsung A33 5G - SDK 34

**Reply by bijington** [reply]
Thanks. @jfversluis i recall there was an issue in the XCT around rotation issues and you applied a fix for it. Do you recall what it was?

**Reply by jfversluis** [reply]
Looking at the issues and PRs there we tried but never 100% fixed it

**Reply by RcBrgr** [reply]
Thanks for the clarification. Since the CameraView is now using CameraX for Android, I might give it a try implementing. Android has a promising documentation: https://developer.android.com/media/camera/camerax/orientation-rotation

**Reply by bijington** [reply]
Good find! Let us know how that goes

**Comment by tschbc** [comment]
@bijington @jfversluis I'm also experiencing this on iOS 17.5.1. Do you know of any workarounds I could use?

**Reply by bijington** [reply]
I suspect what you might be seeing on iOS is a result of there being more than 4 orientations for an iOS device https://developer.apple.com/documentation/uikit/uideviceorientation. I have seen in the past that even the iOS camera app can fail to get the photo orientation exactly correct when the device is close to `faceUp` or `faceDown`. I am not aware of anything that can be done immediately, I think in the past I have monitored the last used good orientation (excluding `faceUp` and `faceDown`) and used that as a best guess.

**Reply by RcBrgr** [reply]
I have a solution implemented for iOS and Android, needs a bit more testing but it is promising. I need to wait currently for approval of the CLA from my employer, then I will open a PR.

**Reply by tschbc** [reply]
> I suspect what you might be seeing on iOS is a result of there being more than 4 orientations for an iOS device https://developer.apple.com/documentation/uikit/uideviceorientation. I have seen in the past that even the iOS camera app can fail to get the photo orientation exactly correct when the device is close to `faceUp` or `faceDown`. I am not aware of anything that can be done immediately, I think in the past I have monitored the last used good orientation (excluding `faceUp` and `faceDown`) and used that as a best guess.

@bijington I'm probably experiencing that too, but even before that, if I take a picture in any orientation other than portrait the pictures will be rotated. Thank you for the Apple docs link, I'll see if I can get that to work.

When you get the orientation info, what do you do to apply it? Rotate the image data itself? Add an EXIF tag?

> I have a solution implemented for iOS and Android, needs a bit more testing but it is promising. I need to wait currently for approval of the CLA from my employer, then I will open a PR.

@RcBrgr Have they responded yet?

**Reply by RcBrgr** [reply]
@tschbc unfortunately not, things like that can take quite a while.

**Reply by tschbc** [reply]
@RcBrgr without getting into code, are you willing/able to share broad strokes of your solution?

I've been experimenting with ImageSharp and it's been working well on Windows, but glacially slow on iOS.

**Comment by tschbc** [comment]
Unsure if this was already known, but when checking iOS CameraView's photos' EXIF tags, orientation is always set to `upper-right`.

**Reply by bijington** [reply]
If that's the case it sounds like a bug!

**Reply by tschbc** [reply]
I think I'll file an issue for that then! I'll make a repro repo.

**Reply by bijington** [reply]
Thank you! That would be really helpful

**Reply by tschbc** [reply]
@bijington Created an issue https://github.com/CommunityToolkit/Maui/issues/2064

**Comment by CliffAgius** [comment]
Closed as issue #2064 created, if closed in error please let us know to re-open.
